bug: CHECKOV crashes without output in super-linter for repos with large JSON files

## Problem

The CHECKOV linter in the super-linter workflow crashes silently in repositories that contain large generated JSON files (e.g., OpenAPI specifications). The super-linter logs show:

```
[ERROR]   Errors found in CHECKOV
```

But **no stdout or stderr is captured** — no check results, no CKV violation IDs, no stack traces. The CHECKOV group opens and closes with only the error marker.

## Evidence

- **Failing run (api-specs-enriched PR #115):** https://github.com/f5xc-salesdemos/api-specs-enriched/actions/runs/24597337382
  - CHECKOV section has zero output between `##[group]CHECKOV` and `##[endgroup]`
  - 22 other super-linter checks pass (pylint, mypy, ruff, biome, yaml, markdown, etc.)
  - The repo contains a 38MB `openapi.json` master spec and 40+ domain spec JSON files

- **Passing run (same repo, governance sync PR #111):** https://github.com/f5xc-salesdemos/api-specs-enriched/actions/runs/24595771880
  - CHECKOV passes with full output: `Passed checks: 300, Failed checks: 0, Skipped checks: 1`
  - This PR had a smaller file changeset

- **Local checkov (v3.2.521) passes clean:** `Passed checks: 300, Failed checks: 0, Skipped checks: 0` across all frameworks (github_actions, secrets, sca_package)

## Hypothesis

CHECKOV 3.2.513 in the super-linter Docker container appears to crash (likely OOM or timeout) when scanning repositories with large JSON files. The crash occurs before any check results are written, which is why there's no stdout. The `.checkov.yaml` config skips specific checks but doesn't prevent checkov from loading/parsing all files in the workspace.

## Suggested Next Steps

1. **Add logging/diagnostics to the super-linter workflow** to capture checkov's stderr and exit code separately, so future crashes produce actionable output instead of silent failures
2. **Investigate resource limits** in the super-linter Docker container — checkov may need more memory or a longer timeout when scanning large repos
3. **Determine if checkov scans non-infrastructure files** — if it's loading 38MB JSON specs that aren't Terraform/CloudFormation/Kubernetes manifests, that's wasted work that could trigger the crash

## Environment

- super-linter: v8.6.0
- checkov: 3.2.513 (inside super-linter container)
- Repository: f5xc-salesdemos/api-specs-enriched
- Large files: `docs/specifications/api/openapi.json` (38MB), `docs/specifications/api/virtual.json` (6MB), plus 36 other generated spec JSONs

--- a/.checkov.yaml
+++ b/.checkov.yaml
@@ -1,10 +1,29 @@
 ---
+# ── Path exclusions ──────────────────────────────────────────────────────────
+# Directories that never contain IaC. Avoids parsing large generated/vendored
+# files that cause memory exhaustion (e.g. 88 MB of OpenAPI spec JSON).
 skip-path:
   - dist
   - build
   - release
   - vendor
+  - node_modules
+  - docs/specifications
+  - specs/discovered
+  - specs/raw
+
+# ── Framework restriction ────────────────────────────────────────────────────
+# Only run frameworks relevant to this org's repos. Eliminates ~35 frameworks
+# that parse every file looking for CloudFormation/ARM/Kubernetes/etc. patterns
+# in repos that contain none of those.
+framework:
+  - github_actions
+  - github_configuration
+  - secrets
+  - terraform
+
+# ── Check exclusions ─────────────────────────────────────────────────────────
 skip-check:
-  - CKV_GHA_7
-  - CKV2_GHA_1
+  - CKV_GHA_7    # GitHub Actions — allow pinning to branch refs
+  - CKV2_GHA_1   # GitHub Actions — immutable actions (covered by Dependabot)
   - "CKV_SECRET_4:.*specifications/api/.*\\.json$"

